site/app/controllers/admin/PlagiarismController.php

```diff
@@ -345,15 +345,11 @@ public function saveNewPlagiarismConfiguration($new_or_edit, $gradeable_id = nul
             $old_array = json_decode($old_config, true);
             $regex_in_old = in_array("regex", $old_array);
             $regex_in_new = in_array("regex", $json_data);
-            $json_data["regex_updated"] = false;
-            if (
-                ($regex_in_old
-                && !$regex_in_new)
-                 || (!$regex_in_old
-                 && $regex_in_new)
-                 || ($old_array['regex'] != $json_data['regex'])
-            ) {
+            if ($regex_in_old || $regex_in_new) {
+                $json_data["regex_updated"] = false;
```

Comment:
Is there harm of moving this assignment into where we construct the `$json_data` object on line 326 - 339?

Reply:
I'm not sure how we would move the logic to lines 326-339.  Don't we still need to check if there is a regex?